### PR TITLE
py-pytest: add 7.3.2

### PR DESCRIPTION
**Pull request by manuelakuhn**
https://github.com/pytest-dev/pytest/tree/7.3.2

### CODE REVIEW

**Comment by manuelakuhn** [comment]
@spackbot fix style

File: var/spack/repos/builtin/packages/py-pytest/package.py [hunk 1]
```diff
@@ -35,10 +37,6 @@ class PyPytest(PythonPackage):
 
     # python_requires
     depends_on("python@3.7:", when="@7.1:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pytest/package.py:39]: 7.3.2 is also the first version to add Python 3.12 support if you want to code that in: https://github.com/pytest-dev/pytest/releases/tag/7.3.2

File: var/spack/repos/builtin/packages/py-pytest/package.py [hunk 2]
```diff
@@ -72,6 +67,9 @@ class PyPytest(PythonPackage):
     depends_on("py-tomli@1:", when="@7.0", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pytest/package.py:69]: importlib-metadata could be swapped in order

**Review by adamjstewart — DISMISSED** [review]

File: var/spack/repos/builtin/packages/py-pytest/package.py (hunk 1 above)
Comment (manuelakuhn) [reply var/spack/repos/builtin/packages/py-pytest/package.py:39]: How would I do this? Isn't that already included by `python@3.7:`

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-pytest/package.py:39]: I'm thinking something like:
```python
depends_on("python@:3.11", when="@:7.3.1", type=("build", "run"))
```

Comment (manuelakuhn) [reply var/spack/repos/builtin/packages/py-pytest/package.py:39]: Ah, ok, thanks. Now I understand what you mean.
Done.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**